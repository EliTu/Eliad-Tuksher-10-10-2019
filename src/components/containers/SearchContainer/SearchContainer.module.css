.SearchContainerStyles {
    width: 40%;
    height: 4rem;
    border: 2px solid black;
}