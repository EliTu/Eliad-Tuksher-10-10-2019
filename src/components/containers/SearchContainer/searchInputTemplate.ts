export const inputTemplateData = {
	elementType: 'input',
	elementConfig: {
		type: 'search',
		placeholder: 'Please enter a city name',
		label: '',
		min: 1,
		max: 30,
	},
	value: '',
	validation: {
		required: false,
		valid: true,
		hasUserInput: false,
		errorMessage: 'Input should have at least one character long',
	},
	isFocused: false,
};
