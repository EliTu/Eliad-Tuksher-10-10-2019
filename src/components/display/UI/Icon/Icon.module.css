.IconStyles {
    /* margin: 0 auto;
    border: 2px solid black;
    padding: 1rem;
    display: flex; */
}