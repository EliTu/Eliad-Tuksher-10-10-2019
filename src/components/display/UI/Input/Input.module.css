@import './../../../../variables.css';

.InputStyles {
    width: 98%;
    box-sizing: border-box;
    font-family: inherit;
}

.InputStyles input,
textarea {
    background-color: rgba(241, 238, 238, 0);
    color: var(--color-main-dark);
    font-size: 1.2rem;
    width: 100%;
    height: 100%;
    padding: 0.3rem 1rem;
    margin: 0;
    display: inline-block;
    transition: all 0.5s;
    outline: none;
}

.validationStyles {}

.invalidStyle {}

.errorMessageStyle {}