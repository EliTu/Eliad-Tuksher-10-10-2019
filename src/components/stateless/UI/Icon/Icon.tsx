import React from 'react';
import styles from './Icon.module.css';
import {
	FontAwesomeIcon,
	FontAwesomeIconProps,
} from '@fortawesome/react-fontawesome';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';

interface IProps {
	iconType: IconDefinition;
	size: FontAwesomeIconProps['size'];
	pulse?: boolean;
	spin?: boolean;
}

const Icon: React.FC<IProps> = ({ iconType, size, pulse, spin }) => {
	const { IconStyles } = styles;

	return (
		<div className={IconStyles}>
			<FontAwesomeIcon
				icon={iconType}
				size={size}
				pulse={pulse}
				spin={spin}
			/>
		</div>
	);
};

export default Icon;
