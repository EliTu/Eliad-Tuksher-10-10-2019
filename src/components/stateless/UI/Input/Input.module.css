@import './../../../../variables.css';

.InputStyles {
    width: 100%;
    box-sizing: border-box;
    font-family: inherit;
}

.InputStyles input,
textarea {
    background-color: rgba(241, 238, 238, 0);
    color: var(--color-main-dark);
    font-size: 1.2rem;
    width: 100%;
    height: 100%;
    border: 2px solid;
    border-top-left-radius: 8px;
    border-bottom-left-radius: 8px;
    border-color: var(--color-main-dark);
    padding: 0.3rem 2rem;
    margin: 0;
    display: inline-block;
    transition: all 0.5s;
}

.validationStyles {}

.invalidStyle {}

.errorMessageStyle {}