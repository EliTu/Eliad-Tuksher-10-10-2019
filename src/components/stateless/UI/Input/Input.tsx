import React, { useRef, useEffect } from 'react';
import styles from './Input.module.css';
import { IInputProps } from './inputPropsInterface';

const Input: React.FC<IInputProps> = ({
	elementType,
	elementConfig,
	value,
	validation,
	isFocused,
	handleChange,
	handleEnterPress,
}) => {
	const { InputStyles, invalidStyle, ValidStyle, errorMessageStyle } = styles;

	let inputElement = null;

	let errorMessageElement: JSX.Element = (
		<p className={errorMessageStyle}>{validation.errorMessage}</p>
	);

	// Listen to keyboard enter click to submit form:
	const enterPressCallback = (event: any, func: (e: Event) => void) => {
		if (event.key === 'Enter') func(event);
	};

	// Focus the first input field upon component mount
	const focusRef: React.RefObject<any> = useRef();

	useEffect(() => {
		if (isFocused) focusRef.current.focus();
	}, [isFocused]);

	// Set validation styles:
	let validationStyles: string = '';
	validationStyles =
		!validation.valid && validation.hasUserInput
			? invalidStyle
			: validation.valid && validation.hasUserInput && value !== ''
			? ValidStyle
			: '';

	switch (elementType) {
		case 'input':
			inputElement = (
				<input
					ref={focusRef}
					className={validationStyles}
					{...elementConfig}
					value={value}
					onChange={handleChange}
					onKeyPress={event =>
						enterPressCallback(event, handleEnterPress)
					}
					data-test="input-test"
				/>
			);
			break;

		case 'textarea':
			inputElement = (
				<textarea
					{...elementConfig}
					value={value}
					onChange={handleChange}
					onKeyPress={event =>
						enterPressCallback(event, handleEnterPress)
					}
					data-test="textarea-test"
				/>
			);
			break;

		case 'select':
			inputElement = (
				<select
					value={value}
					onChange={handleChange}
					data-test="select-test"
				>
					{elementConfig.options &&
						elementConfig.options.map(option => (
							<option
								value={option.value}
								key={option.value}
								disabled={option.disabled}
							>
								{option.displayValue}
							</option>
						))}
				</select>
			);
			break;

		default:
			inputElement = (
				<input
					{...elementConfig}
					value={value}
					// className={!validation.valid ? InvalidStyle : null}
					onKeyPress={event =>
						enterPressCallback(event, handleEnterPress)
					}
				/>
			);
	}
	return (
		<div className={InputStyles}>
			<label>{elementConfig.label}</label>
			{inputElement}
			{errorMessageElement}
		</div>
	);
};

export default Input;
