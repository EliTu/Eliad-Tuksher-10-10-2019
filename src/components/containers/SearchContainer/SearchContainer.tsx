import React, { useState } from 'react';
import Input from '../../stateless/UI/Input/Input';
import { inputTemplateData } from './searchInputTemplate';
import styles from './SearchContainer.module.css';

const SearchContainer: React.FC = () => {
	const { SearchContainerStyles } = styles;

	const [inputData, setInputData] = useState(inputTemplateData);

	const handleSearchInputChange = (event: any) => {
		setInputData(() => (inputData.value = event.target.value));
	};
	const handleSearchSubmission = () => {};

	return (
		<div className={SearchContainerStyles}>
			<Input
				elementType={inputData.elementType}
				elementConfig={{ ...inputData.elementConfig }}
				value={inputData.value}
				validation={{ ...inputData.validation }}
				isFocused={inputData.isFocused}
				handleChange={event => handleSearchInputChange(event)}
				handleEnterPress={handleSearchSubmission}
			/>
		</div>
	);
};

export default SearchContainer;
