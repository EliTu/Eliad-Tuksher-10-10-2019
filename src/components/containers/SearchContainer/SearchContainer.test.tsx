import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { SearchContainer } from './SearchContainer';
import Input from '../../display/UI/Input/Input';
import Icon from '../../display/UI/Icon/Icon';
import initialShallowRender from '../../../utilities/test-utilities/initialShallowRender';

describe('SearchContainer Component', () => {
	let component: ShallowWrapper;
	beforeEach(
		() => (component = shallow(<SearchContainer httpRequest={() => {}} />))
	);

	it('should render without errors', () =>
		initialShallowRender(component, '.SearchContainerStyles'));

	it('should render a div wrapper that contains an Input and an Icon components', () => {
		expect(component.children().find('div').length).toBe(1);

		expect(component.children().find(Input)).toExist();
		expect(component.children().find(Input).length).toBe(1);

		expect(component.children().find(Icon)).toExist();
		expect(component.children().find(Icon).length).toBe(1);
	});
});
