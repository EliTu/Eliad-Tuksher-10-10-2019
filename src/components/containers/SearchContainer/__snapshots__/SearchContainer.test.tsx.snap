// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer Component should render without errors 1`] = `
<div
  className="SearchContainerStyles"
>
  <Input
    data="search"
    elementConfig={
      Object {
        "label": "",
        "max": 30,
        "min": 1,
        "placeholder": "Please enter a city name",
        "type": "search",
      }
    }
    elementType="input"
    handleChange={[Function]}
    handleEnterPress={[Function]}
    isFocused={false}
    validation={
      Object {
        "errorMessage": "Input should have at least one character long",
        "hasUserInput": false,
        "required": false,
        "valid": true,
      }
    }
    value=""
  />
</div>
`;
